site/app/models/User.php

```diff
@@ -436,8 +448,17 @@ private function setDisplayedLastName() {
         $this->displayed_last_name = (!empty($this->preferred_last_name)) ? $this->preferred_last_name : $this->legal_last_name;
     }
 
-    public function getDisplayFullName() {
-        return $this->getDisplayedFirstName() . ' ' . $this->getDisplayedLastName();
+    public function getDisplayFullName(): string {
+        $first_name = $this->getDisplayedFirstName();
+        $last_name = $this->getDisplayedLastName();
+        switch ($this->preferred_name_order) {
+            case self::FIRST_LAST:
+                return $first_name . ' ' . $last_name;
+            case self::LAST_FIRST:
+                return $last_name . ' ' . $first_name;
+            default:
+                return $first_name . ' ' . $last_name;
+        }
```

Comment:
We could simplify this a bit:
```suggestion
        switch ($this->preferred_name_order) {
            case self::LAST_FIRST:
                return $last_name . ' ' . $first_name;
            case self::FIRST_LAST:
            default:
                return $first_name . ' ' . $last_name;
        }
```